---
title: "Small RNA Counts Analysis"
author: "Andrew Burr"
date: "September 14, 2016"
output: html_document
---

```{r setup, include=FALSE}

require(limma)
require(knitr)
require(edgeR)
require(biomaRt)
require(DESeq2)
require(edgeR)
require(ggplot2)
require(reshape)
require(reshape2)
require(stringdist)
require(apcluster)
require(qqman)
require(Biostrings)
require(motifRG)
require(stringr)
require(scales)
require(ggrepel)


conversion <- read.csv("../../reference_sets/ensembl_to_hgnc.txt", stringsAsFactors=FALSE)
options(scipen=999)


#create a fasta file from a list of sequences
createpiRNAFasta <- function(fileName, extension){
  
  #read in the old file
  fileText <- readLines(con  =fileName)
  
  newText <-vector(length = length(fileText)*2)
  
  #change name
  n <- 1
  for(line in seq(1,length(newText),2)){
    #the fasta sequence to be inverted
    sequence <- fileText[n]
    #go to each letter and switch it
    newText[line] <- paste(">",sequence, sep = "") 
    n <- n + 1
  }
  
  n <- 1
  for(line in seq(2,length(newText),2)){
    sequence <- fileText[n]
    #go to each letter and switch it
    newText[line] <- sequence
    n <- n + 1
  }
  
  return(newText)
  #write.table(derp, "test.fa", sep = "\t", row.names = FALSE, col.names = FALSE, quote = FALSE)
}



```


**piRNA Read Curation**

Below are the steps to create the putative piRNA counts. All read counts from the 25-33 nucleotide subset were aligned agaisnt various genomes. The aligned reads for rRNA  other small non-coding RNA were removed.

```{r piRNA_read_curation}



#import the desired read counts
read_counts <- read.delim("../sRNA_counts/small_sa_og_norRNA_25_33_counts_table.txt", sep=",", stringsAsFactors=FALSE, row.names = 1)[,]
#select only the 8 paired samples desired for analysis
read_counts <- read_counts[,4:19]

#load in the list of reads the DID align to rRNA
rRNA <- read.table("../data/small_sa_og_norRNA_25_33_rRNA_list.txt", header = FALSE, stringsAsFactors = FALSE)
#remove those reads from the data set
read_counts <- read_counts[!(rownames(read_counts) %in% rRNA$V1),]

#load in the list of reads the DID align to other small non-coding RNA
sRNA <- read.table("../data/small_sa_og_norRNA_25_33_sRNA_list.txt", header = FALSE, stringsAsFactors = FALSE)
#remove those reads from the data set
read_counts <- read_counts[!(rownames(read_counts) %in% sRNA$V1),]

#keep reads found in greater than 1 count in 6 or more samples
read_counts <- read_counts[rowSums(read_counts > 1 ) >= 6,]

#save the reads to a table list
#write.table("../data/small_sa_og_norRNA_25_33_noncRNA_list.txt", quote  = FALSE, row.names  = FALSE, col.names = F)


#create a fasta file for the putative piRNA reads
noncRNA  <- createpiRNAFasta("../data/small_sa_og_norRNA_25_33_noncRNA_list.txt", ".txt")

#write.table(as.data.frame(test), "../fastx_files/small_sa_og_norRNA_25_33_noncRNA_list.fa", quote= F, row.names = F, col.names = F, sep = "\t")



```


**Visualizing Read Counts**

Code useful for seeing how many reads survives each pre-processing step.


```{r total_reads, echo = FALSE}


total_counts <- read.table("../sRNA_counts/small_R1_all_counts_table.txt", quote="\"", comment.char="", stringsAsFactors=FALSE, sep = ",", header = TRUE)
#only need stem and diff samples
total_counts <- total_counts[!(total_counts$line %in%  c("4121_piwil1", "4121_piwil2", "4121_IP", "hNP1_old", "hNP1_new", "17231", "16157", "1123++","528++","1123-", "528-")),]

total_counts <- melt(total_counts,id.vars = 13)

total_counts$line <- factor(total_counts$line)
total_counts$Phenotype <- factor(rep(c(rep("stem", 8), rep("diff", 8)), nrow(total_counts)/16))



#All reads NOT in 25-33 length

ggplot(data = total_counts[total_counts$variable %in% c("total", "sa", "sa_og", "sa_og_norRNA"),], aes(x = line)) +
  geom_bar(position = "dodge",stat = 'identity', aes(y = value, fill = variable), color = 'black') +
  labs(title = "Small RNA Read Counts") +
  ylab("Number of Reads in Group") +
  xlab("Cell Line") +
  scale_y_continuous(labels = comma) +
   theme_bw() +
    theme(axis.text.x = element_text(angle = 90),
        panel.grid.major.x = element_blank(),
        panel.grid.minor.x = element_blank()) 





#Reads in the 25-33 length 


#pdf(file = "../../figures/Total_Counts_boxplot.pdf", 6,4)

#boxplot ~ phenotype
ggplot(data = total_counts[total_counts$variable %in% c( "sa_og_norRNA_25_33", "sa_og_norRNA_25_33_noncRNA", "sa_og_norRNA_25_33_noncRNA_repeat",    "sa_og_norRNA_25_33_noncRNA_norepeat"),]) +
  #geom_bar(position = "dodge",stat = 'identity', aes(y = value, fill = variable,x = line), color = 'black') +
  geom_boxplot(aes(fill =Phenotype, y = value, x =  variable  )) +
  labs(title = "Small RNA Read Counts") +
  ylab("Number of Reads in Group") +
  xlab("sRNA Class (25-33 nt)") +
  scale_y_continuous(labels = comma ) +
 scale_fill_brewer(name = "Phenotype", labels = c("GSC", "Non-GSC"), breaks = c("stem", "diff")) +   theme_bw() +
  theme(axis.text.x = element_text(angle = 0 ),
        panel.grid.major.x = element_blank(),
        panel.grid.minor.x = element_blank()) +
  scale_x_discrete(labels = c("Total sRNA", "Putative \npiRNA", "Repeat \n Derived \n piRNA", "Unique \nMapping \n piRNA"))


#dev.off()



```


**Known piRNA**

here, the code finds how many reads in each sample align to a known piRNA sequence.


```{r known_piRNA}



#import the desired read counts
read_counts <- read.delim("../sRNA_counts/small_sa_og_norRNA_25_33_counts_table.txt", sep=",", stringsAsFactors=FALSE, row.names = 1)[,]
read_counts <- read_counts[,4:19]

#select only putative piRNA
noncRNA <- read.table("../data/small_sa_og_norRNA_25_33_noncRNA_list.txt", header = FALSE, stringsAsFactors = FALSE)
read_counts <- read_counts[(rownames(read_counts) %in% noncRNA$V1),]


#laod in the total counts for each processing step
total_counts <- read.table("../sRNA_counts/small_R1_all_counts_table.txt", quote="\"", comment.char="", stringsAsFactors=FALSE, sep = ",", header = TRUE)
#selecting only the stem diff samples
total_counts <- total_counts[!(total_counts$line %in%  c("4121_piwil1", "4121_piwil2", "4121_IP", "hNP1_old", "hNP1_new", "17231", "16157", "1123++","528++","1123-", "528-")),] 
#the filters are what sequences the piRNA align to
filters <- list.files("../data/known_piRNA/", full.names = TRUE)
titles <- c("piRBase", "piRNAdb")




#import the desired read counts
read_counts <- read.delim("../sRNA_counts/small_sa_og_norRNA_25_33_counts_table.txt", sep=",", stringsAsFactors=FALSE, row.names = 1)[,]
read_counts <- read_counts[,4:19]
#keep only reads in the putative piRNA list
noncRNA <- read.table("../data/small_sa_og_norRNA_25_33_noncRNA_list.txt", header = FALSE, stringsAsFactors = FALSE)
read_counts <- read_counts[(rownames(read_counts) %in% noncRNA$V1),]




temp <- as.data.frame(matrix(nrow = 2, ncol =1, byrow = T,data =  c(NA,NA)) )
colnames(temp) <- "Total Reads"
rownames(temp) <- c("Uniquely\nMapping\npiRNA", "RD piRNA")


#join all the biogenesis counts together
for(n in 1:2){
  
  
  
  unique <- read.table("../data/small_sa_og_norRNA_25_33_list_unique_mapping_genome_v0_m1.txt", header = FALSE, stringsAsFactors = FALSE)
  temp_counts <- read_counts[(rownames(read_counts) %in% unique$V1),]
  temp[1,1] <- nrow(temp_counts)
  
  
  filter <- read.table(filters[n], header = FALSE, stringsAsFactors = FALSE)
  
  database_counts <- temp_counts[rownames(temp_counts) %in% filter$V1,]
  temp[1,titles[n]] <- nrow(database_counts)
  
}



#join all the biogenesis counts together
for(n in 1:2){
  
  
  
  unique <- read.table("../data/small_sa_og_norRNA_25_33_list_unique_mapping_genome_v0_m1.txt", header = FALSE, stringsAsFactors = FALSE)
  temp_counts <- read_counts[!(rownames(read_counts) %in% unique$V1),]
  temp[2,1] <- nrow(temp_counts)
  
  
  filter <- read.table(filters[n], header = FALSE, stringsAsFactors = FALSE)
  
  database_counts <- temp_counts[!rownames(temp_counts) %in% filter$V1,]
  temp[2,titles[n]] <- nrow(database_counts)
  
}



#bar plot of unique reads found in previous databases

temp$piRNA <- rownames(temp)
temp <- melt(data = temp, id.vars = c(4))


#pdf("../../figures/Database_curated_piRNA.pdf", 6,4)

#ggplot(data = temp[(temp$variable %in% c("3' UTR", "cDNA", "Repeat \nSequences", "lincRNA")), ]) +
ggplot(data = temp) + 
  #geom_boxplot(aes(y = value, x = variable, fill = cell_type)) +
  geom_bar(stat = 'identity', aes( y = value, x= variable ,fill = piRNA), position = "dodge") + 
  #labs(title = "piRNA Counts from Predicted Biogenesis Locations") +
  ylab("Unique piRNA") +
  xlab("piRNA Database") +
  scale_y_continuous(labels = comma, breaks = c(0,500,1000,1500,2000,2500,3000,3500,4000,4500,5000,5500)) +
  #scale_fill_brewer(name = "Phenotype", labels = c("Stem", "Non-\nstem-\nlike"), breaks = c("stem", "diff")) +
  scale_fill_manual(values = c("grey", "black")) +
  #scale_color_brewer(values = c(""))
  #scale_x_discrete(angle = 45) +
  
  theme_bw() +
  theme(
    axis.text.x = element_text(angle = 0),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank()

  )


#dev.off()


  

```


**Predicted Biogenesis**

Code showing counts for predicted sources of piRNA biogenesis.

```{r biogenesis_stats}



#import the desired read counts
read_counts <- read.delim("../sRNA_counts/small_sa_og_norRNA_25_33_counts_table.txt", sep=",", stringsAsFactors=FALSE, row.names = 1)[,]
read_counts <- read_counts[,4:19]

noncRNA <- read.table("../data/small_sa_og_norRNA_25_33_noncRNA_list.txt", header = FALSE, stringsAsFactors = FALSE)
read_counts <- read_counts[(rownames(read_counts) %in% noncRNA$V1),]



total_counts <- read.table("../sRNA_counts/small_R1_all_counts_table.txt", quote="\"", comment.char="", stringsAsFactors=FALSE, sep = ",", header = TRUE)
#selecting only the stem diff samples
total_counts <- total_counts[!(total_counts$line %in%  c("4121_piwil1", "4121_piwil2", "4121_IP", "hNP1_old", "hNP1_new", "17231", "16157", "1123++","528++","1123-", "528-")),] 
filters <- list.files("../data/biogenesis_lists/", full.names = TRUE)
titles <- c("3' UTR", "cDNA", "CDS","Repeat \nSequences", "lincRNA", "piRNAdb", "RefSeq \n Genomic")



  #create a dataframe of the biogenesis
  df <- as.data.frame(matrix(nrow = 16, ncol = 1, total_counts$sa_og_norRNA_25_33_noncRNA_norepeat, byrow = TRUE))
  colnames(df) <- "total_piRNA"
  df$cell_line <- c("387++","456++","3359++","3691++","3832++","4121++","4302++","H2S++","387-","456-","3359-","3691-","3832-","4121-","4302-","H2S-")
  df$cell_type <- factor(c(rep("stem", 8), (rep("diff",8))))
  #df$RP_piRNA <- total_counts$sa_og_norRNA_25_33_noncRNA_repeat

   temp <- df
    df$piRNA <- "Unique piRNA"
    temp$piRNA <- "RD piRNA"
    df$total_piRNA <- total_counts$putative_piRNA
    temp$total_piRNA <- total_counts$putative_RP_piRNA
  
  
  #join all the biogenesis counts together
  for(n in 1:7){
    
   
    
    unique <- read.table("../data/small_sa_og_norRNA_25_33_list_unique_mapping_genome_v0_m1.txt", header = FALSE, stringsAsFactors = FALSE)
    temp_counts <- read_counts[(rownames(read_counts) %in% unique$V1),]

    filter <- read.table(filters[n], header = FALSE, stringsAsFactors = FALSE)
    #print(file)
    column_sums <- as.data.frame(colSums(temp_counts[rownames(temp_counts) %in% filter$V1,]))
    rownames(column_sums) <- c("387++","456++","3359++","3691++","3832++","4121++","4302++","H2S++","387-","456-","3359-","3691-","3832-","4121-","4302-","H2S-")
    colnames(column_sums) <- titles[n]
      

  
    df <- cbind(df,column_sums)
    
  }
   
  for(n in 1:7){ 
    
    unique <- read.table("../data/small_sa_og_norRNA_25_33_list_unique_mapping_genome_v0_m1.txt", header = FALSE, stringsAsFactors = FALSE)
    temp_counts <- read_counts[!(rownames(read_counts) %in% unique$V1),]

    filter <- read.table(filters[n], header = FALSE, stringsAsFactors = FALSE)
    #print(file)
    column_sums <- as.data.frame(colSums(temp_counts[rownames(temp_counts) %in% filter$V1,]))
    rownames(column_sums) <- c("387++","456++","3359++","3691++","3832++","4121++","4302++","H2S++","387-","456-","3359-","3691-","3832-","4121-","4302-","H2S-")
    colnames(column_sums) <-titles[n]
    
    
    temp <- cbind(temp,column_sums)
  }
  
  df <- rbind(df, temp)
  

  
#  Counts of piRNA from each predicted biogenesis location based on PHENOTYPE
  
 # pdf("../../figures/Boxplot_biogenesis_phenotype.pdf", 6, 4)


temp <- melt(data = df[,-c(1,4)], id.vars = c(1,2))

temp <- aggregate(data = temp, value ~ cell_line + cell_type+variable, sum)


ggplot(data = temp[(temp$variable %in% c("3' UTR", "cDNA", "Repeat \nSequences", "lincRNA")), ]) +
#ggplot(data = temp) + 
  geom_boxplot(aes(y = log(value,2), x = variable, fill = cell_type)) +
  labs(title = "piRNA Counts from Predicted Biogenesis Locations") +
  ylab("Log 2Counts of piRNA") +
  xlab("piRNA Predicted Biogenesis") +
  scale_y_continuous(labels = comma) +
  scale_fill_brewer(name = "Phenotype", labels = c("GSC", "Non-GSC"), breaks = c("stem", "diff")) +
  
  theme_bw() +
  theme(
    axis.text.x = element_text(angle = 0),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank()
  )




#dev.off()


#counts of piRNA from predicted biogenesis based on piRNA TYPE

#pdf("../../figures/Boxplot_biogenesis_piRNAtype.pdf", 6, 4)



temp <- df
temp <- melt(data = temp[,], id.vars = c(2,3,4))

temp$value <- temp$value + 1

temp <- aggregate(data = temp, value ~ cell_line + piRNA+variable, sum)


ggplot(data = temp[(temp$variable %in% c("3' UTR", "cDNA", "Repeat \nSequences", "lincRNA")), ]) +
#ggplot(data = temp) + 
  geom_boxplot(aes(y = log(value,2), x = variable, color = piRNA), size = 1.1) +
  #labs(title = "piRNA Counts from Predicted Biogenesis Locations") +
  ylab("Log2 Counts of piRNA") +
  xlab("piRNA Predicted Biogenesis") +
  scale_y_continuous(labels = comma) +
  #scale_fill_manual(name = "piRNA", values = c("grey","black")) +
  scale_color_manual(values = c("grey", "black"), breaks = c("RD piRNA", "Unique piRNA"), labels = c("RD piRNA", "Unique \n Mapping \n piRNA")) +
  #scale_color_brewer(values = c(""))
  #scale_x_discrete(angle = 45) +
  
  theme_bw() +
  theme(
    axis.text.x = element_text(angle = 0),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank()
    #legend.position = c(.9, .2),
    #legend.background = element_blank(),
     # legend.key.height = unit(x = .1, "cm"),
    #legend.key.width = unit(x = .05, "cm")
  )



#dev.off()




```


**Predicted Biogenesis**

Code showing counts for predicted piRNA targets.

```{r targeting_stats}



#import the desired read counts
read_counts <- read.delim("../sRNA_counts/small_sa_og_norRNA_25_33_counts_table.txt", sep=",", stringsAsFactors=FALSE, row.names = 1)[,]
read_counts <- read_counts[,4:19]


    
    
noncRNA <- read.table("../data/small_sa_og_norRNA_25_33_noncRNA_list.txt", header = FALSE, stringsAsFactors = FALSE)
read_counts <- read_counts[(rownames(read_counts) %in% noncRNA$V1),]

#selecting only the stem diff samples
total_counts <- total_counts[!(total_counts$line %in%  c("4121_piwil1", "4121_piwil2", "4121_IP", "hNP1_old", "hNP1_new", "17231", "16157", "1123++","528++","1123-", "528-")),] 
filters <- list.files("../data/targeting_lists/", full.names = TRUE)
titles <- c( "3' UTR","cDNA", "CDS", "Repeat \n Sequences", "RefSeq \n Geneomic", "Promoter")




  #create a dataframe of the biogenesis
  df <- as.data.frame(matrix(nrow = 16, ncol = 1, total_counts$sa_og_norRNA_25_33_noncRNA_norepeat, byrow = TRUE))
  colnames(df) <- "total_piRNA"
  df$cell_line <- c("387++","456++","3359++","3691++","3832++","4121++","4302++","H2S++","387-","456-","3359-","3691-","3832-","4121-","4302-","H2S-")
  df$cell_type <- factor(c(rep("stem", 8), (rep("diff",8))), levels= c("stem", "diff"))
  #df$RP_piRNA <- total_counts$sa_og_norRNA_25_33_noncRNA_repeat

   temp <- df
    df$piRNA <- "Unique piRNA"
    temp$piRNA <- "RD piRNA"
    df$total_piRNA <- total_counts$putative_piRNA
    temp$total_piRNA <- total_counts$putative_RP_piRNA
  
  
  #join all the biogenesis counts together
  for(n in 1:6){
    
   
    unique <- read.table("../data/small_sa_og_norRNA_25_33_list_unique_mapping_genome_v0_m1.txt", header = FALSE, stringsAsFactors = FALSE)
    temp_counts <- read_counts[(rownames(read_counts) %in% unique$V1),]
    #remove low counts
    temp_counts <- temp_counts[rowSums(temp_counts > 1) >= 6,]
    
    
    
    
    #shorten the reads
    temp_counts$short <- strtrim(rownames(temp_counts),21)
    temp_counts <- aggregate(. ~ short, data = temp_counts, FUN = sum, na.rm = TRUE)
    rownames(temp_counts) <- temp_counts$short
    temp_counts <- subset(temp_counts, select = -short)
    
    
    filter <- read.table(filters[n], header = FALSE, stringsAsFactors = FALSE)
    #print(file)
    column_sums <- as.data.frame(colSums(temp_counts[rownames(temp_counts) %in% filter$V1,]))
    rownames(column_sums) <- c("387++","456++","3359++","3691++","3832++","4121++","4302++","H2S++","387-","456-","3359-","3691-","3832-","4121-","4302-","H2S-")
    colnames(column_sums) <- titles[n]
      

  
    df <- cbind(df,column_sums)
    
  }
   
  for(n in 1:6){ 
       
    unique <- read.table("../data/small_sa_og_norRNA_25_33_list_unique_mapping_genome_v0_m1.txt", header = FALSE, stringsAsFactors = FALSE)
    temp_counts <- read_counts[!(rownames(read_counts) %in% unique$V1),]
    #remove low counts
    temp_counts <- temp_counts[rowSums(temp_counts > 1) >= 6,]
    
    
    
    
    #shorten the reads
    temp_counts$short <- strtrim(rownames(temp_counts),21)
    temp_counts <- aggregate(. ~ short, data = temp_counts, FUN = sum, na.rm = TRUE)
    rownames(temp_counts) <- temp_counts$short
    temp_counts <- subset(temp_counts, select = -short)
    
    
    filter <- read.table(filters[n], header = FALSE, stringsAsFactors = FALSE)
    #print(file)
    column_sums <- as.data.frame(colSums(temp_counts[rownames(temp_counts) %in% filter$V1,]))
    rownames(column_sums) <- c("387++","456++","3359++","3691++","3832++","4121++","4302++","H2S++","387-","456-","3359-","3691-","3832-","4121-","4302-","H2S-")
    colnames(column_sums) <-titles[n]
      

    
  
    temp <- cbind(temp,column_sums)
    
  }
  
  
  df <- rbind(df, temp)
  
  
#plots
  
# Counts based on phenotype and piRNA type
temp <- melt(data = df[,-c(1)], id.vars = c(1,2,3))


#pdf("../../figures/Boxplot_count_target_predictions.pdf", 6, 6)

ggplot(data = temp[(temp$variable %in% c("3' UTR", "cDNA", "Repeat \n Sequences")), ]) +
  geom_boxplot(aes(y = log(value), x = variable, color = piRNA, fill = cell_type)) +
  labs(title = "piRNA with Predicted Target") +
  ylab("Log2 Counts of piRNA") +
  xlab("Predicted Target") +
  #scale_y_continuous(labels = comma, breaks = c(0, 25000, 50000, 75000, 100000, 125000, 150000)) +
    scale_color_manual(values = c("grey", "black"), breaks = c("RD piRNA", "Unique piRNA"), labels = c("RD piRNA", "Unique\nMapping\npiRNA")) +
scale_fill_brewer(name = "Phenotype", labels = c("GSC", "Non-GSC"), breaks = c("stem", "diff")) +                  
  #scale_color_brewer(values = c(""))
  #scale_x_discrete(angle = 45) +
  
  theme_bw() +
  theme(
    axis.text.x = element_text(angle = 0),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank(),
    legend.position = "bottom"
    #legend.position = c(.9, .2),
    #legend.background = element_blank(),
     # legend.key.height = unit(x = .1, "cm"),
    #legend.key.width = unit(x = .05, "cm")
  )

#dev.off()


  
  
  #targets based on phenotype

temp <- melt(data = df[,-c(1,4)], id.vars = c(1,2))

temp <- aggregate(data = temp, value ~ cell_line + cell_type+variable, sum)

#pdf("../../figures/Boxplot_target_phenotype.pdf", 6, 4)



ggplot(data = temp[(temp$variable %in% c("3' UTR", "cDNA", "Repeat \n Sequences", "Promoter")), ]) +
#ggplot(data = temp) + 
 geom_boxplot(aes(y = value, x = variable, fill = cell_type)) +
  labs(title = "Counts of piRNA with Predicted Target") +
  ylab("Counts of piRNA with Biogenesis Sequence") +
  xlab("Unique Mapping piRNA and Repeat Mapping piRNA against Biogenesis Sequence") +
  #scale_y_continuous(labels = comma, breaks = c(0, 25000, 50000, 75000, 100000, 125000, 150000)) +
scale_fill_brewer(name = "Phenotype", labels = c("Stem", "Non-\nstem-\nlike"), breaks = c("stem", "diff")) +   theme_bw() +
  scale_color_manual(values = c("grey", "black")) +
  
  theme_bw() +
  theme(
    axis.text.x = element_text(angle = 0),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank()
  )

  
  
#dev.off()



#counts based on piRNA type

temp <- melt(data = df[,-c(1,3)], id.vars = c(1,2))

#pdf("../../figures/Boxplot_count_target_predictions.pdf", 6, 4)

ggplot(data = temp[(temp$variable %in% c("3' UTR", "cDNA", "Repeat \n Sequences", "Promoter")), ]) +
  geom_boxplot(aes(y = value, x = variable, fill = piRNA)) +
  labs(title = "Boxplot of Counts of piRNA with Predicted Target") +
  ylab("Counts of piRNA with Biogenesis Sequence") +
  xlab("Unique Mapping piRNA and Repeat Mapping piRNA against Biogenesis Sequence") +
  #scale_y_continuous(labels = comma, breaks = c(0, 25000, 50000, 75000, 100000, 125000, 150000)) +
  scale_fill_manual(name = "Phenotype",
                     values = c("darkblue","lightblue")) +
  scale_color_manual(values = c("grey", "black")) +
  #scale_color_brewer(values = c(""))
  #scale_x_discrete(angle = 45) +
  
  theme_bw() +
  theme(
    axis.text.x = element_text(angle = 0),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank()
  )

#dev.off()






#predicted targets based on percentage of total reads


temp <- subset(df, select = c(total_piRNA, piRNA, `3' UTR`, cDNA, Promoter))
temp[,c(3,4,5)] <- (temp[,c(3,4,5)]/temp[,1])*100
#PERCENTAGE BOXPLOTS
temp <- melt(data = temp[,-c(1)], id.vars = c(1))


ggplot(data = temp[(temp$variable %in% c("3' UTR", "cDNA", "Promoter")), ]) +
#ggplot(data = temp) + 
 geom_boxplot(aes(y = value, x = variable, color = piRNA)) +
  labs(title = "Percentages of piRNA with Predicted Target") +
  ylab("Counts of piRNA with Biogenesis Sequence") +
  xlab("Target Sequence") +
  #scale_y_continuous(labels = comma, breaks = c(0, 25000, 50000, 75000, 100000, 125000, 150000)) +
  scale_fill_manual(name = "Phenotype",
                     values = c("darkblue","lightblue")) +
  scale_color_manual(values = c("grey", "black")) +
  #scale_color_brewer(values = c(""))
  #scale_x_discrete(angle = 45) +
  
  theme_bw() +
  theme(
    axis.text.x = element_text(angle = 0),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank()
    #legend.position = c(.9, .2),
    #legend.background = element_blank(),
     # legend.key.height = unit(x = .1, "cm"),
    #legend.key.width = unit(x = .05, "cm")
  )


#targets based on percentage of total reads for PHENOTYPE and pIRNA TYPE

temp <- subset(df, select = c(total_piRNA,cell_type, piRNA, `3' UTR`, cDNA, Promoter))
temp[,c(4,5,6)] <- (temp[,c(4,5,6)]/temp[,1])*100
#PERCENTAGE BOXPLOTS
temp <- melt(data = temp[,-c(1)], id.vars = c(1,2))


#pdf("../../figures/Boxplot_percentage_target_phenotype_piRNA.pdf", 6, 6)

ggplot(data = temp[(temp$variable %in% c("3' UTR", "cDNA", "Promoter")), ]) +
#ggplot(data = temp) + 
 geom_boxplot(aes(y = value, x = variable, color = piRNA, fill = cell_type)) +
  labs(title = "Percentages of piRNA with Predicted Target") +
  ylab("Counts of piRNA with Biogenesis Sequence") +
  xlab("Target Sequence") +
  #scale_y_continuous(labels = comma, breaks = c(0, 25000, 50000, 75000, 100000, 125000, 150000)) +
  scale_fill_manual(name = "Phenotype",
                     values = c("darkblue","lightblue"), labels = c("non-stem \n GBM", "GSC")) +
  scale_color_manual(values = c("black", "grey")) +
  #scale_color_brewer(values = c(""))
  #scale_x_discrete(angle = 45) +
  
  theme_bw() +
  theme(
    axis.text.x = element_text(angle = 0),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank(),
    legend.position = "bottom"
    #legend.background = element_blank(),
     # legend.key.height = unit(x = .1, "cm"),
    #legend.key.width = unit(x = .1, "cm")
  )




#dev.off()





```

**piRNA Found in CLusters**

Code to find piRNAidentified in a cluster.

```{r piRNA_from_clusters, eval=FALSE}

#load the proTRAC files
protrac_folder <- "../proTRAC_files/clusters_minimum_1000/stem_diff"
file_names <- list.files(protrac_folder)
file_names <- strtrim(file_names, 9)


files <- list.files(protrac_folder, full.names = TRUE, recursive = TRUE, pattern = ".fasta")
files <- files[-grep(files, pattern = "clusters.fasta")]


piRNAfromClusters <- function(file){
  fasta <- readDNAStringSet(file)
  return(as.character(fasta, use.names = FALSE))

}

piRNA <- lapply(X = files, FUN =piRNAfromClusters)
piRNA <- unlist(piRNA)
piRNA <- unique(piRNA)
#piRNA <- unlist(lapply(piRNA, FUN = strtrim,21))
piRNA <- unique(piRNA)


read_counts <- read_counts[rownames(read_counts) %in% piRNA,]
```






